### PR TITLE
kotlin-webpack-plugin now works with webpack 4

### PR DESCRIPTION
**Pull request by aduryagin**
* Fix for compileIfKotlinFilesChanged. Webpack now uses Map instead Object in compilation.fileTimestamps.
* Use webpack-log instead custom log function.
* Moved to new webpack 4 api for hooks.
* Minimal supported nodejs version to 8.6.0
* Code improvements.
* 2.0.0

Fix for issue: https://youtrack.jetbrains.com/issue/CRKA-94

### CODE REVIEW

File: packages/kotlin-webpack-plugin/plugin.js [hunk 1]
```diff
@@ -76,16 +77,19 @@ class KotlinWebpackPlugin {
     );
     this.optimizeDeadCode = this.optimizeDeadCode.bind(this);
     this.setPastDate = this.setPastDate.bind(this);
+    this.log = this.log.bind(this);
 
     this.startTime = Date.now();
     this.prevTimestamps = {};
```
[the remark this answers is not in the document]
Comment (aduryagin) [reply packages/kotlin-webpack-plugin/plugin.js]: Yep. I updated the PR.

**Comment by huston007** [comment]
Will this changes still work with previous versions of Webpack (like 2 or 3) or it is breaking change?

**Comment by aduryagin** [comment]
@huston007 No. In v3 and older versions plugin with this changes won't works. Webpack 3 uses old 0.2.7 version of https://github.com/webpack/tapable. New webpack 4 uses new version of tapable 1.1.0. So, it breaking changes and plugin need to be upgraded to v2. I added to readme.md minimal requirements and upgraded plugin to v2 in package.json

**Comment by huston007** [comment]
Thank you so much for your work! We'll test it for some time in a separate branch

**Pull request merged**